### DISCUSSION TITLE
When i change the charaters sprites and play the game it locks up

### DISCUSSION THREAD

**Discussion by Raptorq1** [Techokami/SonicWorldsNext#81] [Q&A]
All I did is change the sprite2d node to animatedsprite2d and when I ran the game it would lock up

When I did this in a older build it wouldn't lock up

**Comment by DimensionWarped** [comment]
Post the change you made or we can't even begin to help.

First though... have you considered *not* doing that? I can't think of a reason why you would want to.

**Reply by Raptorq1** [reply]
I did it because the sprites I am using aren't in a spritesheet

**Reply by DimensionWarped** [reply]
You are just trying to set yourself up for failure at that point. Look, I'm sure this isn't going to be anything resembling easy without a computer, but you are almost certainly going to want to copy whatever frames you are trying to use on top of the old sprite sheet. I'd recommend doing this in something like paint.net or photoshop.

**Reply by Raptorq1** [reply]
But the sprites I am using are from sonic mania

**Reply by DimensionWarped** [reply]
And?

**Reply by Raptorq1** [reply]
The sprites have more frames of animation

**Reply by DimensionWarped** [reply]
Ok, so you're going to need to modify the animation to accomodate that. That's no reason not to use a sprite sheet anyway. Feel free to leave the old run cycle as is and just add the new one to the end of the spritesheet. You can also enlarge the sprite sheet if you need to, but be aware that you'll need to modify the division of the sprite sheet and keep to using an integer multiplier of the existing frame size if you don't want things to get awkward quickly.

It's an awful lot of work mind you and time that could probably be better spent actually working on a level or something that will help your game stick out a lot more than simply using someone else's spritework, but if that's your prerogative have fun.
